src/components/fields/image-upload/image-input/file-item/file-item.tsx

```diff
@@ -0,0 +1,171 @@
+import { Icon } from "@lifesg/react-design-system/icon";
+import { Text } from "@lifesg/react-design-system/text";
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import { FileHelper, TestHelper } from "../../../../../utils";
+import { ERROR_MESSAGES } from "../../../../shared";
+import { EImageStatus, IImage, IImageUploadValidationRule, TImageUploadAcceptedFileType } from "../../types";
+import {
+	CellDeleteButton,
+	CellFileSize,
+	CellInfo,
+	CellProgressBar,
+	DeleteButton,
+	ErrorText,
+	MobileFileSize,
+	ProgressBar,
+	TextBody,
+	Thumbnail,
+	Wrapper,
+} from "./file-item.styles";
+
+interface IProps {
+	id?: string;
+	index: number;
+	fileItem: IImage;
+	maxSize: number;
+	accepts: TImageUploadAcceptedFileType[];
+	validation: IImageUploadValidationRule[];
+	onDelete: (index: number) => (event: React.MouseEvent<HTMLButtonElement>) => void;
+}
+
+export const FileItem = ({ id = "file-item", index, fileItem, maxSize, accepts, onDelete, validation }: IProps) => {
+	// =============================================================================
+	// CONST, STATE, REFS
+	// =============================================================================
+	const { dataURL, file, name: fileName, status, uploadProgress } = fileItem;
+	const [isError, setError] = useState<boolean>(false);
+	const [errorMessage, setErrorMessage] = useState<string>();
+	const fileNameWrapperRef = useRef<HTMLDivElement>(null);
+	const [transformedFileName, setTransformedFileName] = useState<string>();
+	// =============================================================================
+	// HELPER FUNCTIONS
+	// =============================================================================
+	const setFileNameToWidth = useCallback(() => {
+		let widthOfElement = 0;
+		if (fileNameWrapperRef && fileNameWrapperRef.current) {
+			widthOfElement = fileNameWrapperRef.current.getBoundingClientRect().width;
+		}
+		const transformed = FileHelper.truncateFileName(fileName, widthOfElement);
+		setTransformedFileName(transformed);
+	}, [fileName]);
+
+	useEffect(() => {
+		window.addEventListener("resize", setFileNameToWidth);
+		return () => window.removeEventListener("resize", setFileNameToWidth);
+	}, [setFileNameToWidth]);
+
+	// =============================================================================
+	// EFFECTS
+	// =============================================================================
+	useEffect(() => {
```

Comment:
error message for various status is derived from the validation config in the schema, otherwise it will fall back to the default error message